### PR TITLE
Recommend DNS Labels for workload controller names

### PR DESCRIPTION
**Pull request by thockin**
@sftim - you suggested this in https://github.com/kubernetes/kubernetes/pull/114412#issuecomment-1352350183 - help me craft wording?  Once we are happy with words I will apply the same pattern to other types.

### CODE REVIEW

File: content/en/docs/concepts/workloads/controllers/deployment.md [hunk 1]
```diff
@@ -45,6 +45,14 @@ The following is an example of a Deployment. It creates a ReplicaSet to bring up
 In this example:
 
 * A Deployment named `nginx-deployment` is created, indicated by the `.metadata.name` field.
+
+  {{< note >}}
+  The `.metadata.name` field will ultimately be used to name the Pods which are created.  This field allows a
+  [DNS subdomain](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names) value, but this can
+  produce unexpected results for the Pod hostnames.  Using a
+  [DNS label](/docs/concepts/overview/working-with-objects/names#dns-label-names) for the name is recommended.
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/deployment.md:53]:
```suggestion
  produce unexpected results for the Pod hostnames. The Kubernetes project recommends
  that you create Deployments where the `.metadata.name` conforms to the more restrictive
  rules for a
  [DNS label](/docs/concepts/overview/working-with-objects/names#dns-label-names)`
```

File: content/en/docs/concepts/workloads/controllers/deployment.md [hunk 2]
```diff
@@ -45,6 +45,14 @@ The following is an example of a Deployment. It creates a ReplicaSet to bring up
 In this example:
 
 * A Deployment named `nginx-deployment` is created, indicated by the `.metadata.name` field.
+
+  {{< note >}}
+  The `.metadata.name` field will ultimately be used to name the Pods which are created.  This field allows a
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/deployment.md:50]:
```suggestion
  When the control plane creates new Pods for a Deployment, the `.metadata.name` of the
  Deployment   is part of the basis for naming those Pods. For a Deployment, the
  `.metadata.name` field allows a
```

**Review by sftim — COMMENTED** [review]
/lgtm

We could do more here; the overall page needs a tidy up, but that can happen another time.

File: content/en/docs/concepts/workloads/controllers/deployment.md (hunk 1 above)
Comment (thockin) [reply content/en/docs/concepts/workloads/controllers/deployment.md:53]: ACK

**Comment by thockin** [comment]
Updated other types - PTAL

File: content/en/docs/concepts/workloads/controllers/replicationcontroller.md [hunk 3]
```diff
@@ -115,8 +115,14 @@ specifies an expression with the name from each pod in the returned list.
 ## Writing a ReplicationController Spec
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/replicationcontroller.md:115]: (optional)
This heading is increasingly misnamed. I'd be happy to also:
```suggestion
## Writing a ReplicationController manifest
```

File: content/en/docs/concepts/workloads/controllers/cron-jobs.md [hunk 4]
```diff
@@ -36,11 +36,16 @@ for parsing and calculating the next Job creation time. Any usage of it is not
...
-maximum length of a Job name is no more than 63 characters.
+When the control plane creates new Pods for a CronJob, the `.metadata.name` of the
+CronJob is part of the basis for naming those Pods.  The name of a CronJob must be a valid
+[DNS subdomain](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names)
+value, but this can produce unexpected results for the Pod hostnames.  For best compatibility,
+the name should follow the more restrictive rules for a
+[DNS label](/docs/concepts/overview/working-with-objects/names#dns-label-names).
+Even when the name is a DNS subdomain, the name must be no longer than 52
+characters.  This is because the CronJob controller will automatically append
+11 characters to the job name provided and there is a constraint that the
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/cron-jobs.md:47]:
```suggestion
11 characters to the name you provide, and there is a constraint that the
```

File: content/en/docs/concepts/workloads/controllers/cron-jobs.md [hunk 5]
```diff
@@ -36,11 +36,16 @@ for parsing and calculating the next Job creation time. Any usage of it is not
 recommended in a production cluster.
 {{< /caution >}}
 
-When creating the manifest for a CronJob resource, make sure the name you provide
-is a valid [DNS subdomain name](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).
-The name must be no longer than 52 characters. This is because the CronJob controller will automatically
-append 11 characters to the job name provided and there is a constraint that the
-maximum length of a Job name is no more than 63 characters.
+When the control plane creates new Pods for a CronJob, the `.metadata.name` of the
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/cron-jobs.md:39]:
```suggestion
When the control plane creates new Jobs and (indirectly) new Pods for a CronJob,
the `.metadata.name` of the
```
?

File: content/en/docs/concepts/workloads/controllers/deployment.md [hunk 6]
```diff
@@ -45,6 +45,16 @@ The following is an example of a Deployment. It creates a ReplicaSet to bring up
...
 * A Deployment named `nginx-deployment` is created, indicated by the `.metadata.name` field.
+
+  {{< note >}}
+  When the control plane creates new Pods for a Deployment, the `.metadata.name` of the
+  Deployment is part of the basis for naming those Pods.  For a Deployment, the `.metadata.name`
+  field allows a [DNS subdomain](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names)
+  value, but this can produce unexpected results for the Pod hostnames.  The Kubernetes project recommends
+  that you create Deployments where the `.metadata.name` conforms to the more restrictive rules for a
+  [DNS label](/docs/concepts/overview/working-with-objects/names#dns-label-names).
+  {{< /note >}}
```
Comment (sftim) [inline content/en/docs/concepts/workloads/controllers/deployment.md:56]: I wouldn't mention that here. Maybe a brief note and a page-internal link to `#writing-a-deployment-spec`?

If you do that, you could put a similar brief note and another link near:
```markdown
Notice that the name of the ReplicaSet is always formatted as `[DEPLOYMENT-NAME]-[HASH]`.
```

No need to use a `{{< note >}}` for those brief notes.

**Review by sftim — COMMENTED** [review]
/lgtm

with optional further changes

File: content/en/docs/concepts/workloads/controllers/replicationcontroller.md (hunk 3 above)
Comment (thockin) [reply content/en/docs/concepts/workloads/controllers/replicationcontroller.md:115]: I personally sort of loathe the word "manifest" for this, but it seems to be what we use, so....ACK

File: content/en/docs/concepts/workloads/controllers/cron-jobs.md (hunk 5 above)
Comment (thockin) [reply content/en/docs/concepts/workloads/controllers/cron-jobs.md:39]: ACK

File: content/en/docs/concepts/workloads/controllers/cron-jobs.md (hunk 4 above)
Comment (thockin) [reply content/en/docs/concepts/workloads/controllers/cron-jobs.md:47]: ACK

File: content/en/docs/concepts/workloads/controllers/deployment.md (hunk 6 above)
Comment (thockin) [reply content/en/docs/concepts/workloads/controllers/deployment.md:56]: ACK

**Comment by thockin** [comment]
Thanks for the word tweaking.  New push is up.

File: content/en/docs/concepts/workloads/controllers/replicationcontroller.md (hunk 3 above)
Comment (sftim) [reply content/en/docs/concepts/workloads/controllers/replicationcontroller.md]: I like that we have some name for these things that's not “a YAML”. And, in an ecosystem with an arguably unhealthy number of nautical puns, it does kind of fit.

**Review by sftim — COMMENTED** [review]
Thanks again.
/lgtm

**Comment by tengqm** [comment]
/approve
Thanks.

**Pull request merged**